beebop/app.py

```diff
@@ -147,51 +154,74 @@ def add_files(
 
 
 @app.errorhandler(500)
-def internal_server_error(e) -> json:
+def internal_server_error(e) -> tuple[Response, Literal[500]]:
     """
     :param e: [error]
-    :return json: [error response object]
+    :return Response: [error response object]
     """
+    logger.warning(f"Internal Server Error: {e}")
     return (
         jsonify(
             error=response_failure(
-                {"error": "Internal Server Error", "detail": str(e)}
+                ResponseError(
+                    error="Internal Server Error", detail=str(e.description)
+                )
             )
         ),
         500,
     )
 
 
+@app.errorhandler(400)
+def bad_request(e) -> tuple[Response, Literal[400]]:
+    """
+    :param e: [error]
+    :return Response: [error response object]
+    """
+    logger.warning(f"Bad Request: {e}")
+    return (
+        jsonify(
+            error=response_failure(
+                ResponseError(error="Bad Request", detail=str(e.description))
+            )
+        ),
+        400,
+    )
```

Comment:
It feels like all these error handlers follow a similar pattern (log, then call jsonify and return tuple, and could perhaps be made a little more DRY. Maybe `response_failure` could do the jsonifying too? Maybe for the success case too?